src/redact/index.ts

```diff
@@ -21,3 +24,31 @@ export const addDefaultRedactPathStrings = (
   }
   return redact;
 };
+
+export const addRemovePathsCensor = (
+  redact: ExtendedRedact,
+): ExtendedRedact => {
+  if (
+    !redact ||
+    Array.isArray(redact) ||
+    !redact.removePaths ||
+    redact.removePaths.length === 0
+  ) {
+    return redact;
+  }
+
+  const { paths: redactPaths, removePaths } = redact;
+
+  const redactMap = redactPaths.reduce(
+    (previous, current) => ({ ...previous, [current]: true }),
+    {} as Record<string, boolean>,
+  );
+
+  return redactPaths.length === 0
+    ? { paths: removePaths, remove: true }
+    : {
+        paths: [...redactPaths, ...removePaths],
+        censor: (_value, path) =>
+          redactMap[path.join('.')] ? '[Redacted]' : undefined,
```

Comment:
This seems a bit dicey. How does the redaction package handle the difference between e.g.

```typescript
{ a: { b: 'pii' } }

{ 'a.b': 'pii' }
```

Reply:
I've resolved this issue by refactoring out the key builder and putting in a comprehensive test.

I've used a regular expression as I'm not sure of a better way to work out if the identifier name is out-of-spec.